lms/djangoapps/courseware/views.py

```diff
@@ -395,6 +398,43 @@ def _index_bulk_op(request, course_key, chapter, section, position):
             'reverifications': fetch_reverify_banner_info(request, course_key),
         }
 
+        if settings.FEATURES.get('ENTRANCE_EXAMS', False) and getattr(course, 'entrance_exam_enabled', False):
+            # Check to see if the course is gated on required content (such as an Entrance Exam)
+            # Override the chapter and section in order to show its content.
+            required_content = get_required_content(course, user)
+            if len(required_content):
+                for _chapter in course_module.get_display_items():
+                    # If the chapter is in required_content e.g. entrance exam. then show its content.
+                    if unicode(_chapter.location) in required_content and \
+                            not _chapter.hide_from_toc and \
+                            _chapter.is_entrance_exam:
+
+                        # Entrance exam always contains one hidden subsection.
+                        # that's why we are explicitly indexing subsection.
+                        chapter = _chapter.url_name
+                        if len(_chapter.get_display_items()):
+                            section = _chapter.get_display_items()[0].url_name
+
+                        # Calculate the exam score
+                        exam_key = UsageKey.from_string(course.entrance_exam_id)
```

Comment:
In order to keep this very long view function simplier, can you move lines 419 to 434 into a utility file? Say:

get_entrance_exam_score(user, course)